File: checkers/cwe/tests/suite/LoopLimitTaintChecker/java/incorrect_do_loop_argv.java
import java.util.Scanner;

class A {
    int main()
    {
        Scanner in = new Scanner(System.in);
        int x = in.nextInt();
        int y = x + 1;
        int i = 0;
        do {
            i++;
        } while (i < y);  // Loop here
    }
}

File: checkers/cwe/tests/suite/LoopLimitTaintChecker/incorrect_do_loop_argv.cpp
#include <cstdlib>

int EXTER_ATTACK main(int argc, char** argv)
{
    if (argc < 2 || argv == nullptr || argv[1] == nullptr) {
        return -1;
    }

    int y;
    int x = std::atoi(argv[1]);
    y = x + 1;
    int i = 0;
    do {
        i++;
    } while (i < y);  // Loop here
}
